### DISCUSSION TITLE
Rotary encoder reverse logic

### DISCUSSION THREAD

**Discussion by AndyB117** [davetcc/tcMenu#457] [General]
I use rotary encoder. And I connect pins so that when I turn rotary clockwise than navigate UP to menu tree, counter-clockwise - DOWN. But when I edit values in menu items turning clockwise decreases the value rather than increasing it!
In my opinion this is not logical. People usually associate the concepts UP and INCREASE. DOWN and DECREASE. But not vice versa.
How can I make it so that menu navigation UP and INCREASING values are achieved by turning to one side?

**Comment by davetcc** [comment]
This is away from the standard way that the menu system is intended to work, normally menu item navigation works well with the same direction as editing. Do you intend to use such a non-standard navigation in your system, so that turning the encoder up results in the menu going down?

**Reply by davetcc** [reply]
Or put it another way, I'm trying to understand what you're asking? Is this your intended end state to have the menu go down through items when you turn it "up" or clockwise, and then you wish to reverse the direction for editing? Or are you saying to me this is not intended and asking how it can be fixed?

**Reply by AndyB117** [reply]
I want when turning rotary clockwise go UP through menu items and INCREASE value when editing.
But at present I have UP in menu and DECREASE in editing. DOWN in menu and INCREASE when editing.
Is this intended according to the logic of the library or is there something wrong with me (with the settings or with the hardware)?

**Reply by AndyB117** [reply]
[tcMenuTest.tar.gz](https://github.com/davetcc/tcMenu/files/14622258/tcMenuTest.tar.gz)
That is minimal menu test with generated code. MCU - ESP32s2, tcMenu 4.2.2
I already have reverse logic in it (UP->DECREASE, DOWN->INCREASE)

**Comment by davetcc** [comment]
Actually it turns out that this change is trivial, and I will add it because it is literally a one line change to allow existing functionality for up/down keyboards to work. I'll add it but you'll need to test it yourself and feedback if it does not act as expected, as I will only test the regular case.

**Reply by davetcc** [reply]
See https://github.com/davetcc/tcMenu/issues/458
